crates/bevy_infinite_grid/src/render/grid.wgsl

```diff
@@ -95,14 +95,47 @@ fn fragment(in: VertexOutput) -> FragmentOutput {
 
     out.depth = clip_depth;
 
-    let scale = grid_settings.scale;
-    let coord = plane_coords * scale; // use the scale variable to set the distance between the lines
+    // Perspective scaling
+
+    let camera_distance_from_plane = abs(dot(view.world_position - plane_origin, plane_normal));
+
+    // The base 10 log of the camera distance
+    let log10_distance = log(max(grid_settings.scale, camera_distance_from_plane)) / log(10.);
+
+    // The scaling to be used when the camera projection has perspective
+    let perspective_scaling = pow(10., floor(log10_distance));
+
+
+    // Orthographic scaling
+
+    // The height of the view in world units
+    let view_area_height = 2. / view.projection[1][1];
+
+    // Who knows what it means?
+    let cool_magic_number = 300.;
+    let size = view_area_height / cool_magic_number;
```

Comment:
I've tried my best to document the new shader code but my brain is cooked on this one for now